realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -308,8 +308,28 @@ public void onCall() {
      * {@link io.realm.Realm#cancelTransaction()}. Transactions are used to atomically create, update and delete objects
      * within a Realm.
      * <p>
-     * Before beginning the transaction, {@link io.realm.Realm#beginTransaction()} updates the Realm in the case of
-     * pending updates from other threads.
+     * Writing to a Realm always happens on the latest version of the data. So before beginning a transaction, this
```

Comment:
Before beginning a transaction the Realm instance is updated to the latest version, which included all the changes from the other threads.